pytensor/tensor/slinalg.py

```diff
@@ -1669,6 +1669,98 @@ def block_diag(*matrices: TensorVariable):
     return _block_diagonal_matrix(*matrices)
 
 
+def _to_banded_form(A, kl, ku):
+    """
+    Convert a full matrix A to LAPACK banded form for gbmv.
+
+    Parameters
+    ----------
+    A: np.ndarray
+        (m, n) banded matrix with nonzero values on the diagonals
+    kl: int
+        Number of nonzero lower diagonals of A
+    ku: int
+        Number of nonzero upper diagonals of A
+
+    Returns
+    -------
+    ab: np.ndarray
+        (kl + ku + 1, n) banded matrix suitable for LAPACK
+    """
+    A = np.asarray(A)
+    m, n = A.shape
+    ab = np.zeros((kl + ku + 1, n), dtype=A.dtype, order="C")
+
+    for i, k in enumerate(range(ku, -kl - 1, -1)):
+        col_slice = slice(k, None) if k >= 0 else slice(None, n + k)
+        ab[i, col_slice] = np.diag(A, k=k)
+
+    return ab
```

Comment:
If you want some extra saves:

```suggestion
from numpy import zeros, diag

...
    m, n = A.shape
    ab = zeros((kl + ku + 1, n), dtype=A.dtype, order="C")

    for i, k in enumerate(range(ku, -kl - 1, -1)):
        ab[i, slice(k, None) if k >= 0 else slice(None, n + k)] = diag(A, k=k)

    return ab
```

And inline it...

Reply:
what do you mean by "inline it" ?

Reply:
don't put it in a helper function (functions are costly to call), just put the code directly in `perform`